File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-com-5-4/ElectricityDeliveryType.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class ElectricityDeliveryType : ISerialized
    {
        public ElectricityDeliveryType(XmlNode xmlNode)
        {
            XmlNode firmNode = xmlNode.SelectSingleNode("firm");
            
            if (firmNode != null)
            {
                if (firmNode.Attributes["href"] != null || firmNode.Attributes["id"] != null) 
                {
                    if (firmNode.Attributes["id"] != null) 
                    {
                        firmIDRef_ = firmNode.Attributes["id"].Value;
                        ElectricityDeliveryFirm ob = new ElectricityDeliveryFirm(firmNode);
                        IDManager.SetID(firmIDRef_, ob);
                    }
                    else if (firmNode.Attributes["href"] != null)
                    {
                        firmIDRef_ = firmNode.Attributes["href"].Value;
                    }
                    else
                    {
                        firm_ = new ElectricityDeliveryFirm(firmNode);
                    }
                }
                else
                {
                    firm_ = new ElectricityDeliveryFirm(firmNode);
                }
            }
            
        
            XmlNode nonFirmNode = xmlNode.SelectSingleNode("nonFirm");
            
            if (nonFirmNode != null)
            {
                if (nonFirmNode.Attributes["href"] != null || nonFirmNode.Attributes["id"] != null) 
                {
                    if (nonFirmNode.Attributes["id"] != null) 
                    {
                        nonFirmIDRef_ = nonFirmNode.Attributes["id"].Value;
                        XsdTypeBoolean ob = new XsdTypeBoolean(nonFirmNode);
                        IDManager.SetID(nonFirmIDRef_, ob);
                    }
                    else if (nonFirmNode.Attributes["href"] != null)
                    {
                        nonFirmIDRef_ = nonFirmNode.Attributes["href"].Value;
                    }
                    else
                    {
                        nonFirm_ = new XsdTypeBoolean(nonFirmNode);
                    }
                }
                else
                {
                    nonFirm_ = new XsdTypeBoolean(nonFirmNode);
                }
            }
            
        
            XmlNode systemFirmNode = xmlNode.SelectSingleNode("systemFirm");
            
            if (systemFirmNode != null)
            {
                if (systemFirmNode.Attributes["href"] != null || systemFirmNode.Attributes["id"] != null) 
                {
                    if (systemFirmNode.Attributes["id"] != null) 
                    {
                        systemFirmIDRef_ = systemFirmNode.Attributes["id"].Value;
                        ElectricityDeliverySystemFirm ob = new ElectricityDeliverySystemFirm(systemFirmNode);
                        IDManager.SetID(systemFirmIDRef_, ob);
                    }
                    else if (systemFirmNode.Attributes["href"] != null)
                    {
                        systemFirmIDRef_ = systemFirmNode.Attributes["href"].Value;
                    }
                    else
                    {
                        systemFirm_ = new ElectricityDeliverySystemFirm(systemFirmNode);
                    }
                }
                else
                {
                    systemFirm_ = new ElectricityDeliverySystemFirm(systemFirmNode);
                }
            }
            
        
            XmlNode unitFirmNode = xmlNode.SelectSingleNode("unitFirm");
            
            if (unitFirmNode != null)
            {
                if (unitFirmNode.Attributes["href"] != null || unitFirmNode.Attributes["id"] != null) 
                {
                    if (unitFirmNode.Attributes["id"] != null) 
                    {
                        unitFirmIDRef_ = unitFirmNode.Attributes["id"].Value;
                        ElectricityDeliveryUnitFirm ob = new ElectricityDeliveryUnitFirm(unitFirmNode);
                        IDManager.SetID(unitFirmIDRef_, ob);
                    }
                    else if (unitFirmNode.Attributes["href"] != null)
                    {
                        unitFirmIDRef_ = unitFirmNode.Attributes["href"].Value;
                    }
                    else
                    {
                        unitFirm_ = new ElectricityDeliveryUnitFirm(unitFirmNode);
                    }
                }
                else
                {
                    unitFirm_ = new ElectricityDeliveryUnitFirm(unitFirmNode);
                }
            }
            
        
        }
        
    
        #region Firm_
        private ElectricityDeliveryFirm firm_;
        public ElectricityDeliveryFirm Firm_
        {
            get
            {
                if (this.firm_ != null)
                {
                    return this.firm_; 
                }
                else if (this.firmIDRef_ != null)
                {
                    firm_ = IDManager.getID(firmIDRef_) as ElectricityDeliveryFirm;
                    return this.firm_; 
                }
                else
                {
                      return this.firm_; 
                }
            }
            set
            {
                if (this.firm_ != value)
                {
                    this.firm_ = value;
                }
            }
        }
        #endregion
        
        public string firmIDRef_ { get; set; }
        #region NonFirm_
        private XsdTypeBoolean nonFirm_;
        public XsdTypeBoolean NonFirm_
        {
            get
            {
                if (this.nonFirm_ != null)
                {
                    return this.nonFirm_; 
                }
                else if (this.nonFirmIDRef_ != null)
                {
                    nonFirm_ = IDManager.getID(nonFirmIDRef_) as XsdTypeBoolean;
                    return this.nonFirm_; 
                }
                else
                {
                      return this.nonFirm_; 
                }
            }
            set
            {
                if (this.nonFirm_ != value)
                {
                    this.nonFirm_ = value;
                }
            }
        }
        #endregion
        
        public string nonFirmIDRef_ { get; set; }
        #region SystemFirm_
        private ElectricityDeliverySystemFirm systemFirm_;
        public ElectricityDeliverySystemFirm SystemFirm_
        {
            get
            {
                if (this.systemFirm_ != null)
                {
                    return this.systemFirm_; 
                }
                else if (this.systemFirmIDRef_ != null)
                {
                    systemFirm_ = IDManager.getID(systemFirmIDRef_) as ElectricityDeliverySystemFirm;
                    return this.systemFirm_; 
                }
                else
                {
                      return this.systemFirm_; 
                }
            }
            set
            {
                if (this.systemFirm_ != value)
                {
                    this.systemFirm_ = value;
                }
            }
        }
        #endregion
        
        public string systemFirmIDRef_ { get; set; }
        #region UnitFirm_
        private ElectricityDeliveryUnitFirm unitFirm_;
        public ElectricityDeliveryUnitFirm UnitFirm_
        {
            get
            {
                if (this.unitFirm_ != null)
                {
                    return this.unitFirm_; 
                }
                else if (this.unitFirmIDRef_ != null)
                {
                    unitFirm_ = IDManager.getID(unitFirmIDRef_) as ElectricityDeliveryUnitFirm;
                    return this.unitFirm_; 
                }
                else
                {
                      return this.unitFirm_; 
                }
            }
            set
            {
                if (this.unitFirm_ != value)
                {
                    this.unitFirm_ = value;
                }
            }
        }
        #endregion
        
        public string unitFirmIDRef_ { get; set; }
        public string choiceStr_0;
        
    
        
    
    }
    
}

File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-com-5-4/ElectricityDeliveryType.cpp
// ElectricityDeliveryType.cpp 
#include "ElectricityDeliveryType.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

ElectricityDeliveryType::ElectricityDeliveryType(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //firmNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* firmNode = xmlNode->FirstChildElement("firm");

   if(firmNode){firmIsNull_ = false;}
   else{firmIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- firmNode , address : " << firmNode << std::endl;
   #endif
   if(firmNode)
   {
      if(firmNode->Attribute("href") || firmNode->Attribute("id"))
      {
          if(firmNode->Attribute("id"))
          {
             firmIDRef_ = firmNode->Attribute("id");
             firm_ = boost::shared_ptr<ElectricityDeliveryFirm>(new ElectricityDeliveryFirm(firmNode));
             firm_->setID(firmIDRef_);
             IDManager::instance().setID(firmIDRef_,firm_);
          }
          else if(firmNode->Attribute("href")) { firmIDRef_ = firmNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { firm_ = boost::shared_ptr<ElectricityDeliveryFirm>(new ElectricityDeliveryFirm(firmNode));}
   }

   //nonFirmNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* nonFirmNode = xmlNode->FirstChildElement("nonFirm");

   if(nonFirmNode){nonFirmIsNull_ = false;}
   else{nonFirmIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- nonFirmNode , address : " << nonFirmNode << std::endl;
   #endif
   if(nonFirmNode)
   {
      if(nonFirmNode->Attribute("href") || nonFirmNode->Attribute("id"))
      {
          if(nonFirmNode->Attribute("id"))
          {
             nonFirmIDRef_ = nonFirmNode->Attribute("id");
             nonFirm_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(nonFirmNode));
             nonFirm_->setID(nonFirmIDRef_);
             IDManager::instance().setID(nonFirmIDRef_,nonFirm_);
          }
          else if(nonFirmNode->Attribute("href")) { nonFirmIDRef_ = nonFirmNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { nonFirm_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(nonFirmNode));}
   }

   //systemFirmNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* systemFirmNode = xmlNode->FirstChildElement("systemFirm");

   if(systemFirmNode){systemFirmIsNull_ = false;}
   else{systemFirmIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- systemFirmNode , address : " << systemFirmNode << std::endl;
   #endif
   if(systemFirmNode)
   {
      if(systemFirmNode->Attribute("href") || systemFirmNode->Attribute("id"))
      {
          if(systemFirmNode->Attribute("id"))
          {
             systemFirmIDRef_ = systemFirmNode->Attribute("id");
             systemFirm_ = boost::shared_ptr<ElectricityDeliverySystemFirm>(new ElectricityDeliverySystemFirm(systemFirmNode));
             systemFirm_->setID(systemFirmIDRef_);
             IDManager::instance().setID(systemFirmIDRef_,systemFirm_);
          }
          else if(systemFirmNode->Attribute("href")) { systemFirmIDRef_ = systemFirmNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { systemFirm_ = boost::shared_ptr<ElectricityDeliverySystemFirm>(new ElectricityDeliverySystemFirm(systemFirmNode));}
   }

   //unitFirmNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* unitFirmNode = xmlNode->FirstChildElement("unitFirm");

   if(unitFirmNode){unitFirmIsNull_ = false;}
   else{unitFirmIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- unitFirmNode , address : " << unitFirmNode << std::endl;
   #endif
   if(unitFirmNode)
   {
      if(unitFirmNode->Attribute("href") || unitFirmNode->Attribute("id"))
      {
          if(unitFirmNode->Attribute("id"))
          {
             unitFirmIDRef_ = unitFirmNode->Attribute("id");
             unitFirm_ = boost::shared_ptr<ElectricityDeliveryUnitFirm>(new ElectricityDeliveryUnitFirm(unitFirmNode));
             unitFirm_->setID(unitFirmIDRef_);
             IDManager::instance().setID(unitFirmIDRef_,unitFirm_);
          }
          else if(unitFirmNode->Attribute("href")) { unitFirmIDRef_ = unitFirmNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { unitFirm_ = boost::shared_ptr<ElectricityDeliveryUnitFirm>(new ElectricityDeliveryUnitFirm(unitFirmNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<ElectricityDeliveryFirm> ElectricityDeliveryType::getFirm()
{
   if(!this->firmIsNull_){
        if(firmIDRef_ != ""){
             return boost::shared_static_cast<ElectricityDeliveryFirm>(IDManager::instance().getID(firmIDRef_));
        }else{
             return this->firm_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<ElectricityDeliveryFirm>();
   }
}
boost::shared_ptr<XsdTypeBoolean> ElectricityDeliveryType::getNonFirm()
{
   if(!this->nonFirmIsNull_){
        if(nonFirmIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeBoolean>(IDManager::instance().getID(nonFirmIDRef_));
        }else{
             return this->nonFirm_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeBoolean>();
   }
}
boost::shared_ptr<ElectricityDeliverySystemFirm> ElectricityDeliveryType::getSystemFirm()
{
   if(!this->systemFirmIsNull_){
        if(systemFirmIDRef_ != ""){
             return boost::shared_static_cast<ElectricityDeliverySystemFirm>(IDManager::instance().getID(systemFirmIDRef_));
        }else{
             return this->systemFirm_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<ElectricityDeliverySystemFirm>();
   }
}
boost::shared_ptr<ElectricityDeliveryUnitFirm> ElectricityDeliveryType::getUnitFirm()
{
   if(!this->unitFirmIsNull_){
        if(unitFirmIDRef_ != ""){
             return boost::shared_static_cast<ElectricityDeliveryUnitFirm>(IDManager::instance().getID(unitFirmIDRef_));
        }else{
             return this->unitFirm_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<ElectricityDeliveryUnitFirm>();
   }
}
}

